iphone/cli/lib/process-js-task.js

```
@@ -0,0 +1,374 @@
+	 * @return {Promise}
+	 */
+	processJsFile(filePathAndName) {
+		let file = this.resolveRelativePath(filePathAndName);
+		if (!file) {
+			throw new Error(`Unable to resolve relative path for ${filePathAndName}.`);
+		}
+
+		const bootstrapPath = file.substr(0, file.length - 3);  // Remove the ".js" extension.
+		if (bootstrapPath.endsWith('.bootstrap') && !this.jsBootstrapFiles.includes(bootstrapPath)) {
+			this.jsBootstrapFiles.push(bootstrapPath);
+		}
+
+		const info = this.jsFiles[file];
+		if (this.builder.encryptJS) {
+			if (file.indexOf('/') === 0) {
+				file = path.basename(file);
+			}
+			this.builder.jsFilesEncrypted.push(file); // original name
+			file = file.replace(/\./g, '_');
+			info.dest = path.join(this.builder.buildAssetsDir, file);
+			this.builder.jsFilesToEncrypt.push(file); // encrypted name
+		}
+
+		return new Promise((resolve) => {
+			try {
+				this.builder.cli.createHook('build.ios.copyResource', this.builder, (from, to, cb) => {
+					const originalContents = fs.readFileSync(from).toString();
+
+					const r = {
+						original: originalContents,
+						contents: originalContents,
+						symbols: []
+					};
+
+					this.builder.cli.createHook('build.ios.compileJsFile', this.builder, (r, from, to, cb2) => {
+						// Read the possibly modified file contents
+						const source = r.contents;
+
+						// If the file didn't change from previous run, return early
+						const currentHash = this.generateHash(source);
+						const previousHash = this.data.contentHashes[from];
+						if (previousHash && previousHash === currentHash) {
+							this.builder.unmarkBuildDirFile(to);
+							return cb2();
+						}
+
+						this.transformAndCopy(source, from, to, () => {
+							this.data.contentHashes[from] = currentHash;
+							cb2();
+						});
+					})(r, from, to, cb);
+				})(info.src, info.dest, resolve);
+			} catch (ex) {
+				ex.message.split('\n').forEach(m => this.logger.error(m));
```

So I think this sort of code should only be in the builder itself. This should just throw an Error out and let the builder decide whether it wants to just log it, or log and exit the process forcibly.